cf-pubsub/src/main/java/org/eclipse/californium/pubsub/Converter.java

```diff
@@ -0,0 +1,264 @@
+/*******************************************************************************
+ * Copyright (c) 2018 RISE SICS and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *    Wajd Tohme
+ *    Ahmad Hussaein
+ *    Petr Kocián
+ *    Matias Carlander-Reuterfelt
+ *    Ismail Hilal
+ *    Tuna Gersil
+ *    Zainab Alsaadi
+ *
+ ******************************************************************************/
+
+package org.eclipse.californium.pubsub;
+
+import org.eclipse.californium.core.CoapResponse;
+import org.eclipse.californium.core.WebLink;
+import org.eclipse.californium.core.coap.LinkFormat;
+import org.eclipse.californium.core.coap.MediaTypeRegistry;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+/**
+ * The Converter class is a collection of helper functions that make using Californium {@link WebLink}
+ * and {@link CoapResponse} easier to use. It also holds various uri linting functions
+ * The {@link PubSub} methods return CoapResponse and this helps with using them
+ */
+public class Converter {
+
+    /*helper functions for  CoapResponse */
+
+    /**
+     * creates a Set of WebLinks from the ResponseText of a given CoapResponse
+     *
+     * @param response CoapResponse
+     * @return a Set of WebLink
+     */
+    public static Set<WebLink> getWebLinks(CoapResponse response){
+        return LinkFormat.parse(response.getResponseText());
+    }
+
+    /**
+     * renders a WebLink for each available topic in the ResponseText of a given CoapResponse
+     * This would include the node topics that have child/leaf topics in the returned Set
+     *
+     * @param response CoapResponse
+     * @return a Set of WebLink
+     */
+    public static Set<WebLink> getAllWebLinks(CoapResponse response) {
+        return extractAllWebLinks(getWebLinks(response));
+    }
+
+    /*helper functions for Set<WebLink>*/
+
+    /**
+     * renders an array out of a given Set of WebLinks
+     * useful if you want to access a specific WebLink without casting to an ArrayList
+     *
+     * @param webLinks Set<WebLink>
+     * @return an array of WebLinks
+     */
+    public static WebLink[] getArray(Set<WebLink> webLinks) {
+        return webLinks.toArray(new WebLink[0]);
```

Comment:
Seems to be equivalent to `return new WebLink[0];`

Reply:
This function might actually not be needed if this class is to be added to Californium, as someone using Californium should know how to do toArray properly on a Set of WebLink.